File: csharp/Strategy/Pricing.cs
using System.Collections.Generic;
using System.Linq;

namespace DesignPatterns.Strategy
{
    abstract class Pricing
    {
        abstract public double calculate(List<Equipment> e);
    }

    class OneItemPricing : Pricing
    {
        public override double calculate(List<Equipment> e)
        {
            return e.Sum(r => r.price);
        }
    }

    class TwoItemPricing : Pricing
    {
        public override double calculate(List<Equipment> e)
        {
            return e.Max(r => r.price);
        }
    }

    class BundlePricing : Pricing
    {
        public override double calculate(List<Equipment> e)
        {
            return e.Sum(r => r.price)*0.7;
        }
    }
}

File: java/Strategy/Pricing.java
/**
 * Created by georgef on 6/3/16.
 */
abstract class Pricing {

    abstract double calculate(double price);
}

class ComboPricing extends Pricing {

    public double calculate(double price) {

        return (price * 0.8);
    }
}

class NormalPricing extends Pricing {

    public double calculate(double price) {

        return price;
    }
}
